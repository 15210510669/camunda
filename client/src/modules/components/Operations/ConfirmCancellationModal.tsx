/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';

import Modal, {SIZES} from 'modules/components/Modal';

type Props = {
  onModalClose: () => void;
  isVisible: boolean;
  onApplyClick: () => void;
  instanceId: string;
};

const ConfirmCancellationModal: React.FC<Props> = ({
  onModalClose,
  isVisible,
  onApplyClick,
  instanceId,
}) => (
  <Modal onModalClose={onModalClose} isVisible={isVisible} size={SIZES.SMALL}>
    <Modal.Header>Cancel Instance</Modal.Header>
    <Modal.Body>
      <Modal.BodyText>
        {`About to cancel Instance ${instanceId}. In case there are called instances, these will be canceled too.`}
        <p> Click “Apply” to proceed.</p>
      </Modal.BodyText>
    </Modal.Body>
    <Modal.Footer>
      <Modal.SecondaryButton title="Cancel" onClick={onModalClose}>
        Cancel
      </Modal.SecondaryButton>
      <Modal.PrimaryButton title="Apply" onClick={onApplyClick}>
        Apply
      </Modal.PrimaryButton>
    </Modal.Footer>
  </Modal>
);

export {ConfirmCancellationModal};
