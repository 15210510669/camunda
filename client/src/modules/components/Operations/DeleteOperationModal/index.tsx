/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import Modal, {SIZES} from 'modules/components/Modal';
import * as Styled from './styled';
import {CmButton} from '@camunda-cloud/common-ui-react';

type Props = {
  onModalClose: () => void;
  isVisible: boolean;
  onDeleteClick: React.ComponentProps<typeof CmButton>['onCmPress'];
  instanceId: string;
};

const DeleteOperationModal: React.FC<Props> = ({
  onModalClose,
  isVisible,
  onDeleteClick,
  instanceId,
}) => {
  return (
    <Modal onModalClose={onModalClose} isVisible={isVisible} size={SIZES.SMALL}>
      <Modal.Header>Delete Instance</Modal.Header>
      <Modal.Body>
        <Styled.BodyText>
          <div>About to delete Instance {instanceId}.</div>
          <div>Click "Delete" to proceed.</div>
        </Styled.BodyText>
      </Modal.Body>
      <Modal.Footer>
        <Styled.SecondaryButton title="Cancel" onClick={onModalClose}>
          Cancel
        </Styled.SecondaryButton>
        <CmButton
          appearance="danger"
          label="Delete"
          onCmPress={onDeleteClick}
          data-testid="delete-button"
        />
      </Modal.Footer>
    </Modal>
  );
};

export {DeleteOperationModal};
