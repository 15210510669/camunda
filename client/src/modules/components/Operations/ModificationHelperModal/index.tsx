/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {observer} from 'mobx-react';

import {
  ModificationType,
  Modification,
  AddIcon,
  CancelIcon,
  MoveIcon,
  DiagramLight,
  DiagramDark,
  Container,
  Modifications,
  Checkbox,
} from './styled';
import {currentTheme} from 'modules/stores/currentTheme';
import {getStateLocally, storeStateLocally} from 'modules/utils/localStorage';
import {
  ModalBody,
  ModalFooter,
  ComposedModal,
  Button,
  ModalHeader,
} from '@carbon/react';

type Props = {
  isVisible: boolean;
  onClose: () => void;
};

const ModificationHelperModal: React.FC<Props> = observer(
  ({isVisible, onClose}) => {
    return (
      <ComposedModal
        open={isVisible && !getStateLocally()?.hideModificationHelperModal}
        preventCloseOnClickOutside
        onClose={onClose}
        size="md"
        aria-label="Process Instance Modification Mode"
      >
        <ModalHeader title="Process Instance Modification Mode" />
        <ModalBody tabIndex={0}>
          <Container>
            <p>
              Process instance modification mode allows you to plan multiple
              modifications on a process instance.
            </p>
            <p>
              By clicking on a flow node, you can select one of following
              modifications if applicable:
            </p>
            <Modifications>
              <Modification>
                <ModificationType>
                  Add <AddIcon />
                </ModificationType>
                a single flow node instance
              </Modification>
              <Modification>
                <ModificationType>
                  Cancel <CancelIcon />
                </ModificationType>
                all running flow node instances
              </Modification>
              <Modification>
                <ModificationType>
                  Move <MoveIcon />
                </ModificationType>
                all the running instances to a different target flow node in the
                diagram
              </Modification>
            </Modifications>
            <p>
              Additionally, you add/edit variables by selecting the flow node
              scope in the Instance History panel.
            </p>
            <p>
              A summary of all planned modifications will be shown after
              clicking on “Apply Modifications”. The modification will be
              applied after the confirmation of the summary.
            </p>
            {currentTheme.theme === 'light' ? (
              <DiagramLight />
            ) : (
              <DiagramDark />
            )}
          </Container>
        </ModalBody>
        <ModalFooter>
          <Checkbox
            labelText="Do not show this message again"
            invalidText=""
            warnText=""
            id="do-not-show"
            onChange={(_, {checked}) => {
              storeStateLocally({
                hideModificationHelperModal: checked,
              });
            }}
          />
          <Button kind="primary" onClick={onClose}>
            Continue
          </Button>
        </ModalFooter>
      </ComposedModal>
    );
  },
);

export {ModificationHelperModal};
