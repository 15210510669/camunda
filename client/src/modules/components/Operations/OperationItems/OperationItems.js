/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import PropTypes from 'prop-types';

import {OPERATION_TYPE} from 'modules/constants';
import * as Styled from './styled';

const iconsMap = {
  [OPERATION_TYPE.RESOLVE_INCIDENT]: <Styled.RetryIcon />,
  [OPERATION_TYPE.CANCEL_WORKFLOW_INSTANCE]: <Styled.CancelIcon />
};

export default function OperationItems(props) {
  return (
    <Styled.Ul {...props}>{React.Children.toArray(props.children)}</Styled.Ul>
  );
}

OperationItems.propTypes = {
  children: PropTypes.node.isRequired
};

OperationItems.Item = function Item({title, onClick, type, ...rest}) {
  return (
    iconsMap.hasOwnProperty(type) && (
      <Styled.Li onClick={onClick}>
        <Styled.Button {...rest} type={type} title={title}>
          {iconsMap[type]}
        </Styled.Button>
      </Styled.Li>
    )
  );
};

OperationItems.Item.propTypes = {
  type: PropTypes.oneOf(Object.keys(OPERATION_TYPE)).isRequired,
  onClick: PropTypes.func,
  title: PropTypes.string
};
