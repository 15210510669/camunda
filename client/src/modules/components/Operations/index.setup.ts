/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {createInstance, createOperation} from 'modules/testUtils';

const INSTANCE = createInstance({
  id: 'instance_1',
  operations: [createOperation({state: 'FAILED'})],
  hasActiveOperation: false,
});
const ACTIVE_INSTANCE = createInstance({
  id: 'instance_1',
  operations: [createOperation({state: 'SENT'})],
  hasActiveOperation: true,
});

const mockOperationCreated = {
  id: '2',
  name: null,
  type: 'CANCEL_WORKFLOW_INSTANCE',
  startDate: '2020-09-29T12:32:54.874+0000',
  endDate: null,
  username: 'demo',
  instancesCount: 2,
  operationsTotalCount: 2,
  operationsFinishedCount: 0,
};

export {INSTANCE, ACTIVE_INSTANCE, mockOperationCreated};
