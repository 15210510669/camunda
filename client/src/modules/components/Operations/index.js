/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import PropTypes from 'prop-types';
import {isEqual} from 'lodash';
import {OPERATION_TYPE, OPERATION_STATE} from 'modules/constants';
import {ACTIVE_OPERATION_STATES} from 'modules/constants';
import {operationsStore} from 'modules/stores/operations';
import {instances as instancesStore} from 'modules/stores/instances';
import {observer} from 'mobx-react';

import {
  getLatestOperation,
  isWithIncident,
  isRunning,
} from 'modules/utils/instance';

import OperationItems from 'modules/components/OperationItems';
import {OperationSpinner} from 'modules/components/OperationSpinner';

import * as Styled from './styled';

const Operations = observer(
  class Operations extends React.Component {
    static propTypes = {
      instance: PropTypes.object.isRequired,
      selected: PropTypes.bool,
      onButtonClick: PropTypes.func,
      forceSpinner: PropTypes.bool,
    };

    static defaultProps = {
      forceSpinner: false,
    };

    state = {operationState: '', operationType: ''};

    componentDidMount = () => {
      const {operations} = this.props.instance;
      if (operations.length > 0) {
        const {type, state} = getLatestOperation(operations);
        this.setState({
          operationState: state,
          operationType: type,
        });
      }
    };

    componentDidUpdate = (prevProps) => {
      const {operations} = this.props.instance;
      if (!isEqual(operations, prevProps.instance.operations)) {
        // change operation state & failed operation icons when new page is loaded;
        const {state, type} = getLatestOperation(operations);
        this.setState({
          operationState: state,
          operationType: type,
        });
      } else if (this.props.instance.id !== prevProps.instance.id) {
        // change operation state
        const {state} = getLatestOperation(operations);

        this.setState({operationState: state});
      }
    };

    handleOnClick = async (operationType) => {
      this.setState({operationState: OPERATION_STATE.SCHEDULED});

      operationsStore.applyOperation(this.props.instance.id, {
        operationType,
      });
      this.props.onButtonClick && this.props.onButtonClick(this.props.instance);
    };

    renderItem = (operationType) => {
      const ariaLabelMap = {
        [OPERATION_TYPE.CANCEL_WORKFLOW_INSTANCE]: 'Cancel',
        [OPERATION_TYPE.RESOLVE_INCIDENT]: 'Retry',
      };

      return (
        <OperationItems.Item
          type={operationType}
          onClick={() => this.handleOnClick(operationType)}
          title={`${ariaLabelMap[operationType]} Instance ${this.props.instance.id}`}
        />
      );
    };

    render() {
      const {instance, selected, forceSpinner} = this.props;
      return (
        <Styled.Operations>
          {(forceSpinner ||
            ACTIVE_OPERATION_STATES.includes(this.state.operationState) ||
            instancesStore.state.instancesWithActiveOperations.includes(
              instance.id
            )) && (
            <OperationSpinner
              selected={selected}
              title={`Instance ${instance.id} has scheduled Operations`}
              data-testid="operation-spinner"
            />
          )}
          <OperationItems>
            {isWithIncident(this.props.instance) &&
              this.renderItem(OPERATION_TYPE.RESOLVE_INCIDENT)}
            {isRunning(this.props.instance) &&
              this.renderItem(OPERATION_TYPE.CANCEL_WORKFLOW_INSTANCE)}
          </OperationItems>
        </Styled.Operations>
      );
    }
  }
);

export {Operations};
