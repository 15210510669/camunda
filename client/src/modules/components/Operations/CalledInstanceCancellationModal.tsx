/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import Modal, {SIZES} from 'modules/components/Modal';
import {Link} from 'modules/components/Link';
import {Paths} from 'modules/routes';

type Props = {
  onModalClose: () => void;
  rootInstanceId: string;
  isVisible: boolean;
};

const CalledInstanceCancellationModal: React.FC<Props> = ({
  onModalClose,
  isVisible,
  rootInstanceId,
}) => {
  return (
    <Modal onModalClose={onModalClose} isVisible={isVisible} size={SIZES.SMALL}>
      <Modal.Header>Cancel Instance</Modal.Header>
      <Modal.Body>
        <Modal.BodyText>
          To cancel this instance, the root instance{' '}
          <Link
            to={Paths.processInstance(rootInstanceId)}
            title={`View root instance ${rootInstanceId}`}
          >
            {rootInstanceId}
          </Link>{' '}
          needs to be canceled. When the root instance is canceled all the
          called instances will be canceled automatically.
        </Modal.BodyText>
      </Modal.Body>
      <Modal.Footer>
        <Modal.SecondaryButton title="Close" onClick={onModalClose}>
          Close
        </Modal.SecondaryButton>
      </Modal.Footer>
    </Modal>
  );
};

export {CalledInstanceCancellationModal};
