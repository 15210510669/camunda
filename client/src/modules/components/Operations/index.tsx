/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React, {useState} from 'react';
import {OPERATION_TYPE} from 'modules/constants';
import {operationsStore} from 'modules/stores/operations';
import {instancesStore} from 'modules/stores/instances';
import {observer} from 'mobx-react';
import {isWithIncident, isRunning} from 'modules/utils/instance';
import OperationItems from 'modules/components/OperationItems';
import {OperationSpinner} from 'modules/components/OperationSpinner';
import {ConfirmOperationModal} from 'modules/components/ConfirmOperationModal';
import {OperationsContainer} from './styled';
import {CalledInstanceCancellationModal} from './CalledInstanceCancellationModal';

type Props = {
  instance: ProcessInstanceEntity;
  isSelected?: boolean;
  onOperation?: () => void;
  onFailure?: () => void;
  forceSpinner?: boolean;
};

const Operations: React.FC<Props> = observer(
  ({instance, onOperation, onFailure, forceSpinner, isSelected = false}) => {
    const handleClick = async (
      operationType: InstanceOperationEntity['type']
    ) => {
      operationsStore.applyOperation({
        instanceId: instance.id,
        payload: {
          operationType,
        },
        onError: () => {
          onFailure?.();
        },
      });
      onOperation?.();
    };

    const [isCancellationModalVisible, setIsCancellationModalVisible] =
      useState(false);

    return (
      <OperationsContainer>
        {(forceSpinner ||
          instancesStore.instanceIdsWithActiveOperations.includes(
            instance.id
          )) && (
          <OperationSpinner
            isSelected={isSelected}
            title={`Instance ${instance.id} has scheduled Operations`}
            data-testid="operation-spinner"
          />
        )}
        <OperationItems>
          {isWithIncident(instance) && (
            <OperationItems.Item
              type={OPERATION_TYPE.RESOLVE_INCIDENT}
              onClick={() => handleClick(OPERATION_TYPE.RESOLVE_INCIDENT)}
              title={`Retry Instance ${instance.id}`}
            />
          )}
          {isRunning(instance) && (
            <OperationItems.Item
              type={OPERATION_TYPE.CANCEL_PROCESS_INSTANCE}
              onClick={() => setIsCancellationModalVisible(true)}
              title={`Cancel Instance ${instance.id}`}
            />
          )}
        </OperationItems>
        {instance.parentInstanceId === null ? (
          <ConfirmOperationModal
            bodyText={`About to cancel Instance ${instance.id}. In case there are called instances, these will be canceled too.`}
            onApplyClick={() => {
              setIsCancellationModalVisible(false);
              handleClick(OPERATION_TYPE.CANCEL_PROCESS_INSTANCE);
            }}
            isVisible={isCancellationModalVisible}
            onModalClose={() => setIsCancellationModalVisible(false)}
            onCancelClick={() => setIsCancellationModalVisible(false)}
          />
        ) : (
          <CalledInstanceCancellationModal
            onModalClose={() => setIsCancellationModalVisible(false)}
            isVisible={isCancellationModalVisible}
            parentInstanceId={instance.parentInstanceId}
          />
        )}
      </OperationsContainer>
    );
  }
);

export {Operations};
