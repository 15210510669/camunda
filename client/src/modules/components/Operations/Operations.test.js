/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import {createMockDataManager} from 'modules/testHelpers/dataManager';

import {createInstance, createOperation} from 'modules/testUtils';

import {STATE, OPERATION_STATE, OPERATION_TYPE} from 'modules/constants';

import Operations from './Operations';
import OperationStatus from 'modules/components/OperationStatus';
import OperationItems from './OperationItems';

jest.mock('modules/utils/bpmn');

describe('Operations', () => {
  let mockOperation, mockInstance, onButtonClick;

  it('should match snapshots', () => {
    // when
    mockOperation = createOperation({state: OPERATION_STATE.SCHEDULED});
    mockInstance = createInstance({operations: [mockOperation]});
    let node = shallow(<Operations.WrappedComponent instance={mockInstance} />);
    //then
    expect(node).toMatchSnapshot();
  });

  it('should pass props OperationStatus', () => {
    // when
    mockOperation = createOperation({
      state: OPERATION_STATE.SCHEDULED,
      type: OPERATION_TYPE.CANCEL_WORKFLOW_INSTANCE
    });
    mockInstance = createInstance({operations: [mockOperation]});
    const mockProps = {
      forceSpinner: true,
      onButtonClick: jest.fn()
    };

    const node = shallow(
      <Operations.WrappedComponent instance={mockInstance} {...mockProps} />
    );

    //then
    expect(node.find(OperationStatus).props().operationState).toBe(
      OPERATION_STATE.SCHEDULED
    );

    expect(node.find(OperationStatus).props().operationType).toBe(
      OPERATION_TYPE.CANCEL_WORKFLOW_INSTANCE
    );
    expect(node.find(OperationStatus).props().forceSpinner).toEqual(
      mockProps.forceSpinner
    );
  });

  describe('Operation Buttons', () => {
    it('should render operation buttons for active instance', () => {
      // when
      mockInstance = createInstance({state: STATE.ACTIVE, operations: []});
      const node = shallow(
        <Operations.WrappedComponent instance={mockInstance} />
      );
      const OperationItemsNode = node.find(OperationItems);
      const Button = OperationItemsNode.find(OperationItems.Item);

      // then
      expect(OperationItemsNode).toExist();
      expect(Button.length).toEqual(1);
      expect(Button.props().type).toEqual(
        OPERATION_TYPE.CANCEL_WORKFLOW_INSTANCE
      );
    });

    it('should render operation buttons for instance with incidents', () => {
      // when
      mockInstance = createInstance({state: STATE.INCIDENT, operations: []});
      const node = shallow(
        <Operations.WrappedComponent instance={mockInstance} />
      );
      const OperationItemsNode = node.find(OperationItems);

      // then
      expect(OperationItemsNode).toExist();
      expect(OperationItemsNode.find(OperationItems.Item).length).toEqual(2);
      expect(
        OperationItemsNode.find(OperationItems.Item)
          .at(0)
          .props().type
      ).toEqual(OPERATION_TYPE.RESOLVE_INCIDENT);
      expect(
        OperationItemsNode.find(OperationItems.Item)
          .at(1)
          .props().type
      ).toEqual(OPERATION_TYPE.CANCEL_WORKFLOW_INSTANCE);
    });

    describe('Retry', () => {
      beforeEach(() => {
        mockInstance = createInstance({state: STATE.INCIDENT});
        onButtonClick = jest.fn();
      });
      afterEach(() => {
        jest.clearAllMocks();
      });

      it('should handle retry of instance incident ', async () => {
        //given
        const node = shallow(
          <Operations.WrappedComponent
            instance={mockInstance}
            dataManager={createMockDataManager()}
            onButtonClick={onButtonClick}
          />
        );

        const operationItem = node.find(OperationItems.Item).at(0);

        // when
        operationItem.simulate('click');

        const {dataManager} = node.instance().props;

        expect(dataManager.applyOperation).toBeCalledWith(mockInstance.id, {
          operationType: OPERATION_TYPE.RESOLVE_INCIDENT
        });
        // expect Spinner to appear
        expect(node.find(OperationStatus).props().operationState).toEqual(
          OPERATION_STATE.SCHEDULED
        );

        // expect callback to be called
        expect(onButtonClick).toHaveBeenCalled();
      });
    });

    describe('Cancel', () => {
      beforeEach(() => {
        mockInstance = createInstance({state: STATE.ACTIVE});
        onButtonClick = jest.fn();
      });
      afterEach(() => {
        jest.clearAllMocks();
      });

      it('should handle the cancelation of an instance ', async () => {
        //given
        const node = shallow(
          <Operations.WrappedComponent
            instance={mockInstance}
            dataManager={createMockDataManager()}
            onButtonClick={onButtonClick}
          />
        );
        const operationItem = node.find(OperationItems.Item);

        // when
        operationItem.simulate('click');

        const {dataManager} = node.instance().props;

        // then
        expect(dataManager.applyOperation).toBeCalledWith(mockInstance.id, {
          operationType: OPERATION_TYPE.CANCEL_WORKFLOW_INSTANCE
        });

        // expect Spinner to appear
        expect(node.find(OperationStatus).props().operationState).toEqual(
          OPERATION_STATE.SCHEDULED
        );

        // expect callback to be called
        expect(onButtonClick).toHaveBeenCalled();
      });
    });
  });
});
