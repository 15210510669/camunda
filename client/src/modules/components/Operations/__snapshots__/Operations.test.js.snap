// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operations should match snapshots 1`] = `
<styled.div>
  <OperationStatus
    forceSpinner={false}
    instance={
      Object {
        "activities": Array [
          Object {
            "activityId": "activityId_0",
            "endDate": "2018-10-10T09:20:38.658Z",
            "id": "id_0",
            "startDate": "2018-10-10T09:20:38.658Z",
            "state": "ACTIVE",
          },
        ],
        "bpmnProcessId": "someKey",
        "endDate": null,
        "id": "id_1",
        "operations": Array [
          Object {
            "endDate": "2018-10-10T09:20:38.661Z",
            "errorMessage": "string",
            "startDate": "2018-10-10T09:20:38.661Z",
            "state": "SCHEDULED",
            "type": "RESOLVE_INCIDENT",
          },
        ],
        "sequenceFlows": Array [],
        "startDate": "2018-06-21",
        "state": "ACTIVE",
        "workflowId": "2",
        "workflowName": "someWorkflowName",
        "workflowVersion": 1,
      }
    }
    operationState="SCHEDULED"
    operationType="RESOLVE_INCIDENT"
  />
  <OperationItems>
    <Item
      onClick={[Function]}
      title="Cancel Instance id_1"
      type="CANCEL_WORKFLOW_INSTANCE"
    />
  </OperationItems>
</styled.div>
`;
